src/lib/dao.js

```diff
@@ -677,26 +674,24 @@ foam.CLASS({
       sink.put(obj);
       this.on.put.publish(obj);
 
-      return promise;
+      return Promise.resolve(obj);
```

Comment:
This isn't the exact behaviour we had before.  Before we returned a promise of the sink, and if the sink wasn't provided then we made the sink a promise of the object.

You could make sink optional and then return Promise.resolve(sink || obj);